cdap-kafka/src/main/java/co/cask/cdap/kafka/run/KafkaServerMain.java

```diff
@@ -130,30 +139,16 @@ public void destroy() {
     // Nothing to do
   }
 
-  private Properties generateKafkaConfig(int brokerId, String zkConnectStr, String hostname, int port,
-                                         CConfiguration cConf) {
-    Preconditions.checkState(port > 0, "Port number is invalid.");
-    int numPartitions = cConf.getInt(KafkaConstants.ConfigKeys.NUM_PARTITIONS_CONFIG_DEPRECATED);
-    Preconditions.checkState(numPartitions > 0, "Num partitions should be greater than zero.");
-
+  private Properties generateKafkaConfig(CConfiguration cConf) {
     Properties prop = new Properties();
 
-    prop.setProperty("broker.id", Integer.toString(brokerId));
-    if (hostname != null) {
-      prop.setProperty("host.name", hostname);
-    }
-    prop.setProperty("port", Integer.toString(port));
-    prop.setProperty("zookeeper.connect", zkConnectStr);
-
-    //ultimately want regex to be "^(kafka\\..server\\.*)$" but for now this gets all deprecated and current config keys
-    //since all deprecated key accesses are redirected to the current config keys in deprecatedKeyMap in Configuration
-    Map<String, String> propConfigs = cConf.getValByRegex("^(kafka\\..*)$");
+    Map<String, String> propConfigs = cConf.getValByRegex("^(kafka\\..server\\.*)$");
```

Comment:
@chtyim I believe showed me a simple test case with `getValByRegex()` to find a deprecated key using a non-deprecated one, so that should not be an issue.